iceoryx_posh/test/moduletests/test_popo_trigger.cpp

```diff
@@ -425,3 +463,223 @@ TEST_F(Trigger_test, TriggerIsNotLogicalEqualWhenInvalid)
+
+    EXPECT_TRUE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, InvalidEventBasedTriggerUpdateOriginDoesNotWork)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    sut.invalidate();
+    TriggerClass anotherTriggerClass;
+    sut.updateOrigin(anotherTriggerClass);
+
+    EXPECT_FALSE(sut.isLogicalEqualTo(&anotherTriggerClass, cxx::ConstMethodCallback<bool>()));
+    EXPECT_FALSE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, EventBasedTriggerWithEmptyResetCallCallsErrorHandlerAndIsInvalid)
+{
+    const uint64_t eventId = 0U;
+    const uint64_t uniqueTriggerId = 0U;
+
+    bool hasTerminated = false;
+    iox::Error errorType = iox::Error::kNO_ERROR;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            hasTerminated = true;
+            errorType = error;
+        });
+
+    Trigger sut(EventBasedTrigger,
+                &m_triggerClass,
+                cxx::MethodCallback<void, uint64_t>(),
+                eventId,
+                TriggerClass::callback,
+                uniqueTriggerId);
+
+    EXPECT_TRUE(hasTerminated);
+    EXPECT_THAT(errorType, Eq(iox::Error::kPOPO__TRIGGER_INVALID_RESET_CALLBACK));
+    EXPECT_FALSE(sut.isValid());
+    EXPECT_FALSE(static_cast<bool>(sut));
+}
+
+TEST_F(Trigger_test, EventBasedMovedConstructedWithValidTriggerWorks)
+{
+    Trigger trigger = createValidEventBasedTrigger();
+    Trigger sut{std::move(trigger)};
+
+    EXPECT_TRUE(sut.isValid());
+    EXPECT_TRUE(static_cast<bool>(sut));
+    EXPECT_THAT(sut.getTriggerType(), Eq(TriggerType::EVENT_BASED));
+    EXPECT_TRUE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+
+    EXPECT_FALSE(trigger.isValid());
+    EXPECT_FALSE(static_cast<bool>(trigger));
+    EXPECT_THAT(trigger.getTriggerType(), Eq(TriggerType::INVALID));
+    EXPECT_FALSE(trigger.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, EventBasedMovedAssignedWithValidTriggerWorks)
+{
+    Trigger sut = createValidStateBasedTrigger();
+    Trigger trigger = createValidEventBasedTrigger();
+    sut = std::move(trigger);
+
+    EXPECT_TRUE(sut.isValid());
+    EXPECT_TRUE(static_cast<bool>(sut));
+    EXPECT_THAT(sut.getTriggerType(), Eq(TriggerType::EVENT_BASED));
+    EXPECT_TRUE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+
+    EXPECT_FALSE(trigger.isValid());
+    EXPECT_FALSE(static_cast<bool>(trigger));
+    EXPECT_THAT(trigger.getTriggerType(), Eq(TriggerType::INVALID));
+    EXPECT_FALSE(trigger.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, EventBasedMovedConstructedWithInvalidTrigger)
+{
+    Trigger trigger = createValidEventBasedTrigger();
+    Trigger trigger1 = std::move(trigger);
```

Comment:
You could call `trigger.invalidate()` instead of invalidating it via another move

Reply:
But I thought I invalidate things with different approaches to see that it also works.